tests/test_tunnel_provider.py

```
@@ -134,6 +134,39 @@ def test_tailscale_public_url_rejects_missing_url():
+                "Web": {
+                    "mj-1.taildc7e9e.ts.net:443": {
+                        "Handlers": {"/": {"Proxy": proxy}}
+                    }
+                },
+            }
+        }
+    })
+
+
+def test_tailscale_public_url_reads_json_funnel_config():
+    run = lambda *args, **kwargs: completed(stdout=_foreground_funnel_json())
```

Ruff (0.16.1)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `run` as a `def`

(E731)